slick/src/main/scala/slick/jdbc/PostgresProfile.scala

```diff
@@ -63,25 +63,53 @@ trait PostgresProfile extends JdbcProfile {
-      val VarCharPattern = "^'(.*)'::character varying$".r
-      val TextPattern = "^'(.*)'::text".r
-      val IntPattern = "^\\((-?[0-9]*)\\)$".r
+      /*
+      The default value for numeric type behave different with postgres version
+      PG9.5 - PG9.6:
+       positive default value in int boundary: 1
+       negative default value in int boundary: '-1'::integer
+       positive default value between int boundary and long boundary: '123123214232131312'::bitint
+       negative default value between int boundary and long boundary: '-123123214232131312'::bitint
+       positive default value beyond long boundary: '111111111111111111111111111'::numeric
+       negative default value beyond long boundary: '-111111111111111111111111111'::numeric
+       positive floating: '1.1'::numeric
+       negative floating: '-.1.1'::numeric
+
+      PGX.X to PG9.4:
+       positive default value in int boundary: 1
+       negative default value in int boundary: (-1)
+       positive default value between int boundary and long boundary: 123123214232131312::bitint
+       negative default value between int boundary and long boundary: (-123123214232131312)::bitint
+       positive default value beyond long boundary: 111111111111111111111111111::numeric
+       negative default value beyond long boundary: (-111111111111111111111111111)::numeric
+       positive floating: 1.1
+       negative floating: (-.1.1)
+
+
+       */
+      val NumericPattern = "^['(]?(-?[0-9]+\\.?[0-9]*)[')]?(?:::(?:numeric|bigint|integer))?".r
+      val TextPattern = "^'(.*)'::(?:bpchar|character varying|text)".r
+      val UUIDPattern = "^'(.*)'::uuid".r
       override def default = meta.columnDef.map((_,tpe)).collect{
         case ("true","Boolean")  => Some(Some(true))
         case ("false","Boolean") => Some(Some(false))
-        case (VarCharPattern(str),"String") => Some(Some(str))
         case (TextPattern(str),"String") => Some(Some(str))
-        case (IntPattern(v),"Int") => Some(Some(v.toInt))
-        case (IntPattern(v),"Long") => Some(Some(v.toLong))
-        case ("NULL::character varying","String") => Some(None)
-        case (v,"java.util.UUID") => {
-          if (v.matches("^['\"].*['\"](::uuid)?$")) {
-            val uuid = v.replaceAll("[\'\"]", "") //strip quotes
-                        .stripSuffix("::uuid") //strip suffix
-            Some(Some(java.util.UUID.fromString(uuid)))
-          } else
-            None // The UUID is generated through a function - treat it as if there was no default.
+        case ("NULL::bpchar", "String") => Some(None)
+        case (TextPattern(str),"Char") => str.length match {
+          case 0 => Some(Some(' ')) // Default to one space, as the char will be space padded anyway
+          case 1 => Some(Some(str.head))
+          case _ => None // This is invalid, so let's not supply any default
```

Comment:
So, a match error from invalid is supposed to be caught and handled depending on `ignoreInvalidDefaults` flag.
This fails silently, but if we let it fall through we get logging, nice error messages, and ability to configure our response.

Let's change this in a future PR.